src/graphdb.js

```diff
@@ -52,74 +52,157 @@ function toPascalCase (str) {
     return r.trim();
 }
 
+// Changes every instance of invalid characters in the given label with the following abbreviations
+function replaceCleanseLabel(label) {
+    return label
+        .replaceAll("!", "_ex_")
+        .replaceAll("$", "_dol_")
+        .replaceAll("&", "_amp_")
+        .replaceAll("(", "_op_")
+        .replaceAll(")", "_cp_")
+        .replaceAll(".", "_dot_")
+        .replaceAll(":", "_cn_")
+        .replaceAll("=", "_eq_")
+        .replaceAll("@", "_at_")
+        .replaceAll("[", "_os_")
+        .replaceAll("]", "_cs_")
+        .replaceAll("{", "_oc_")
+        .replaceAll("|", "_vb_")
+        .replaceAll("}", "_cc_")
+        .replaceAll("-", "_hy_");
+}
 
 function graphDBInferenceSchema (graphbSchema, addMutations) {
     let r = '';
+    let invalidNode = false;
     const gdbs = JSON.parse(graphbSchema);
 
     checkForDuplicateNames(gdbs);
 
     gdbs.nodeStructures.forEach(node => {
         // node type
-        let nodeCase = node.label;
-        if (changeCase) {
-            nodeCase = toPascalCase(node.label); 
+        let nodeCase = replaceCleanseLabel(node.label);
+
+        if (changeCase && (node.label !== nodeCase) || changeCase) {
+            r += `type ${toPascalCase(nodeCase)} @alias(property:"${node.label}") {\n`;
+        }
+        else if (node.label !== nodeCase) {
             r += `type ${nodeCase} @alias(property:"${node.label}") {\n`;
-        } else {
+        }
+        else {
             r += `type ${nodeCase} {\n`;
         }
         
         r += '\t_id: ID! @id\n';
 
         node.properties.forEach(property => {
+            let propertyCase = replaceCleanseLabel(property.name);
```

Comment:
If the property name is `id` we can skip the call to replace the special characters:

```
if (property.name === 'id') {
                r += `\tid: ID\n`;
            } else {
                let propertyCase = replaceCleanseLabel(property.name);
                let alias = '';
                if (property.name !== propertyCase) {
                    alias = ` @alias(property: "${property.name}")`;
                }
                r+= `\t${property.name}: ${property.type}${alias}\n`;
            }
```